cmd/syncthing/gui.go

```diff
@@ -1023,6 +1026,45 @@ func (s *apiService) getSystemLogTxt(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *apiService) getSupportBundle(w http.ResponseWriter, r *http.Request) {
+
+	jsonConfig, _ := json.MarshalIndent(s.cfg.RawCopy(), "", "	")
+	createJSONFile("config.json", jsonConfig)
+
+	jsonLog, _ := json.MarshalIndent(s.systemLog.All(), "", "	")
+	createJSONFile("log.json", jsonLog)
+
+	jsonError, _ := json.MarshalIndent(s.guiErrors.All(), "", "	")
+	createJSONFile("error.json", jsonError)
+
+	heapRuntime := fmt.Sprintf("syncthing-heap-%s-%s-%s-%s.pprof", runtime.GOOS, runtime.GOARCH, Version, time.Now().Format("150405")) // hhmmss
+	createJSONFile("heapRuntime.txt", []byte(heapRuntime))
```

Comment:
Not sure what this is. The ticket talks about capturing a heap profile from the runtime, explaining where the memory is allocated. There is probably some way to get a dump summary of the heap too.